### PR TITLE
Fall back to posix_memalign when OS pages are too big

### PR DESCRIPTION
**Pull request by peterzhu2118**
On systems where OS pages are >16KB (e.g. on cygwin on Windows which have 64KB pages), we fall back to `posix_memalign` to allocate heap pages.

### CODE REVIEW

File: gc.c [hunk 1]
```diff
@@ -814,15 +814,20 @@ typedef struct rb_objspace {
 /* default tiny heap size: 16KB */
 #define HEAP_PAGE_ALIGN_LOG 14
 #define CEILDIV(i, mod) (((i) + (mod) - 1)/(mod))
+#define HEAP_PAGE_ALIGN (1 << HEAP_PAGE_ALIGN_LOG)
+#define HEAP_PAGE_SIZE HEAP_PAGE_ALIGN
```
Comment (peterzhu2118) [inline gc.c:818]: These values had to be moved out of the enum because they're used below in a preprocessor macro.

File: gc.c [hunk 2]
```diff
@@ -6088,7 +6093,6 @@ gc_aging(rb_objspace_t *objspace, VALUE obj)
     }
     check_rvalue_consistency(obj);
 
-    objspace->marked_slots++;
```
Comment (tenderlove) [inline gc.c:6091]: Why remove this?

File: gc.c (hunk 1 above)
Comment (tenderlove) [reply gc.c:818]: Can we keep the enum the same, but also assign to a preprocessor macro?  I'm not 100% sure why we're using an enum, but it's probably safer not to change it.

File: gc.c (hunk 2 above)
Comment (peterzhu2118) [reply gc.c:6091]: That's not in this patch I don't think?

**Pull request merged**